# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieChangeEvent.cpp
# Commit: cc3c39aa8fb8
# Full Hash: cc3c39aa8fb8124cca265e84f71a29cbd218b704
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-12 09:23:07
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 2 - CookieStore API - IPDL, r=edgul,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215145
# ==============================================================================

diff -r 2fc9a3d0ad8b -r cc3c39aa8fb8 dom/cookiestore/CookieChangeEvent.cpp
--- a/dom/cookiestore/CookieChangeEvent.cpp	Wed Sep 11 16:26:54 2024 +0000
+++ b/dom/cookiestore/CookieChangeEvent.cpp	Wed Sep 11 16:26:55 2024 +0000
@@ -50,4 +50,30 @@
   return event.forget();
 }
 
+/* static */ already_AddRefed<CookieChangeEvent>
+CookieChangeEvent::CreateForChangedCookie(EventTarget* aEventTarget,
+                                          const CookieListItem& aItem) {
+  RefPtr<CookieChangeEvent> event =
+      new CookieChangeEvent(aEventTarget, nullptr, nullptr);
+
+  event->InitEvent(u"change"_ns, false, false);
+  event->SetTrusted(true);
+
+  event->mChanged.AppendElement(aItem);
+  return event.forget();
+}
+
+/* static */ already_AddRefed<CookieChangeEvent>
+CookieChangeEvent::CreateForDeletedCookie(EventTarget* aEventTarget,
+                                          const CookieListItem& aItem) {
+  RefPtr<CookieChangeEvent> event =
+      new CookieChangeEvent(aEventTarget, nullptr, nullptr);
+
+  event->InitEvent(u"change"_ns, false, false);
+  event->SetTrusted(true);
+
+  event->mDeleted.AppendElement(aItem);
+  return event.forget();
+}
+
 }  // namespace mozilla::dom